assembly/src/parsers/io_ops.rs

```diff
@@ -262,63 +284,63 @@ fn parse_constants(op: &Token) -> Result<Vec<Felt>, ParsingError> {
-        // parse 1 or more hexadecimal values
-        let values = parse_hex_params(op, param_idx, param_str)?;
-        // push each value onto the stack in order
-        for &value in values.iter() {
-            constants.push(value);
-        }
-    } else {
-        // parse 1 decimal value and push it onto the stack
-        let value = parse_decimal_param(op, param_idx, param_str)?;
+    for param_idx in param_idx..=param_count {
+        let value = parse_element_param(op, param_idx)?;
         constants.push(value);
     }
-
     Ok(constants)
 }
 
-fn parse_hex_params(
-    op: &Token,
-    param_idx: usize,
-    param_str: &str,
-) -> Result<Vec<Felt>, ParsingError> {
+fn parse_hex_params(op: &Token, param_str: &str) -> Result<Vec<Felt>, ParsingError> {
+    let param_idx = 1;
     // handle error cases where the hex string is poorly formed
-    let is_single_element = if param_str.len() <= HEX_CHUNK_SIZE {
-        if param_str.len() % 2 != 0 {
-            // parameter string is not a valid hex representation
-            return Err(ParsingError::invalid_param(op, param_idx));
-        }
-        true
-    } else {
-        if param_str.len() % HEX_CHUNK_SIZE != 0 {
-            // hex string doesn't contain a valid number of bytes
-            return Err(ParsingError::invalid_param(op, param_idx));
-        }
-        false
-    };
+    if param_str.len() % HEX_CHUNK_SIZE != 0 {
+        // hex string doesn't contain a valid number of bytes
+        return Err(ParsingError::invalid_param(op, param_idx));
+    }
 
-    // parse the hex string into one or more valid field elements
-    if is_single_element {
-        // parse a single element in hex representation
-        let parsed_param = parse_hex_param(op, param_idx, param_str)?;
-        Ok(vec![parsed_param])
+    // iterate over the multi-value hex string and parse each 8-byte chunk into a valid element
+    (0..param_str.len())
+        .step_by(HEX_CHUNK_SIZE)
+        .map(|i| parse_hex_param(op, param_idx, &param_str[i..i + HEX_CHUNK_SIZE]))
+        .collect()
+}
+
+fn evaluate_type_one(value: Felt) -> Result<Node, ParsingError> {
+    let value_u64 = value.as_int();
+    if value_u64 <= u8::MAX as u64 {
+        Ok(Instruction(PushU8(value_u64 as u8)))
+    } else if value_u64 <= u16::MAX as u64 {
+        Ok(Instruction(PushU16(value_u64 as u16)))
+    } else if value_u64 <= u32::MAX as u64 {
+        Ok(Instruction(PushU32(value_u64 as u32)))
+    } else if value_u64 < Felt::MODULUS {
+        Ok(Instruction(PushFelt(value)))
     } else {
-        // iterate over the multi-value hex string and parse each 8-byte chunk into a valid element
-        (0..param_str.len())
-            .step_by(HEX_CHUNK_SIZE)
-            .map(|i| parse_hex_param(op, param_idx, &param_str[i..i + HEX_CHUNK_SIZE]))
-            .collect()
+        panic!()
+    }
+}
+
+fn evaluate_type_many(values: Vec<Felt>) -> Result<Node, ParsingError> {
```

Comment:
Same comment as before. We should avoid converting to/from `Felt`'s. Instead, we should work with `u64` values and convert to `Felt` when needed.